servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/AbstractStreamingHttpConnection.java

```diff
@@ -100,10 +107,64 @@ public final <T> Publisher<? extends T> transportEventStream(final HttpEventKey<
                 failed(new IllegalArgumentException("Unknown key: " + eventKey));
     }
 
-    private Single<StreamingHttpResponse> makeRequest(final Publisher<Object> flattenedRequest,
+    private Single<StreamingHttpResponse> makeRequest(final HttpRequestMetaData requestMetaData,
+                                                      final Publisher<Object> flattenedRequest,
                                                       @Nullable final FlushStrategy flushStrategy) {
-        return writeAndRead(flattenedRequest, flushStrategy).liftSyncToSingle(new SpliceFlatStreamToMetaSingle<>(
-                this::newSplicedResponse));
+        @Nullable
+        final OnStreamClosedRunnable onStreamClosed = requestMetaData.context().get(OnStreamClosedRunnable.KEY);
```

Comment:
does this need to be lazy? if not plz add a comment clarifying why.

also consider clarifying why we can't lazy-fetch this value if needed in `canel()` below (then it doesn't look as awkward keeping state outside the operator chain).